src/yunohost/app.py

```diff
@@ -494,7 +507,7 @@ def app_install(auth, app, label=None, args=None):
     os.system('cp %s/manifest.json %s' % (app_tmp_folder, app_setting_path))
     os.system('cp -R %s/scripts %s' % (app_tmp_folder, app_setting_path))
     try:
-        if hook_exec(app_tmp_folder + '/scripts/install', args_list) == 0:
+        if hook_exec(app_tmp_folder + '/scripts/install', args=args_list, env=env_dict) == 0:
```

Comment:
It seems that you forgot to pass the `env` argument for the `app_upgrade` action, as you do there for `app_install`.